jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/Content.java

```diff
@@ -866,6 +908,36 @@ static void write(Sink sink, boolean last, String utf8Content, Callback callback
             sink.write(last, ByteBuffer.wrap(utf8Content.getBytes(StandardCharsets.UTF_8)), callback);
         }
 
+        static void write(Sink sink, boolean last, Content.Source source, Callback callback)
+        {
+            Content.Source.Aware aware = findContentSourceAware(sink);
+            if (aware != null)
+            {
+                // Optimization to enable zero-copy.
+                aware.setContentSource(source);
+                sink.write(last, TRANSFER, callback);
```

Comment:
I feel that we should allow some failures on the callback here to fall through to a normal copy.  I.e. if something in the implementation is Aware but cannot do a transferTo (perhaps because it would be non optimal) then it can fail this write with a `TransferToFailed` exception, which would then just do the normal copy below.